[Refactor] 게시글 목록 조회에 필터 추가한 거 수정 + 유저 좋아요 여부, 유저 본인 게시글 여부, 제목/내용으로 검색 기능 추가

2anizirong opened the pull request:
## #️⃣ Related Issues

- resolves #89 

## 🧑‍💻 작업 내용 및 결과

<img width="1273" height="342" alt="image" src="https://github.com/user-attachments/assets/d81d0fc1-47cf-472c-bc23-8e2032c7eae8" />
- 해당 게시글 목록 조회에 필터 추가해서 스웨거에서 토글로 확인할 수 있도록 수정하였습니다.

🚨 추가 수정사항입니다!!

<img width="904" height="132" alt="image" src="https://github.com/user-attachments/assets/b0668c1a-0020-4b9e-b4e3-74d77b4940a3" />
<img width="835" height="767" alt="image" src="https://github.com/user-attachments/assets/6fcab49d-57eb-40b8-9abd-a9782618b20c" />

- 제목이나 내용으로 게시글을 검색할 수 있는 'search' 파라미터 추가하였습니다.

<img width="236" height="68" alt="image" src="https://github.com/user-attachments/assets/dba84f50-fd3d-49a4-821a-a80873d2bf0a" />

- 해당 게시글에 유저 작성 여부 추가하였습니다.
- 해당 게시글에 유저 좋아요 여부 추가하였습니다.

## 💬 리뷰 시 요청사항 (선택)

x

## 📝 Checklist

- [x] Reviewer를 추가했나요?
- [x] Convention을 준수했나요?

2anizirong committed:
♻️ Refactor: 커뮤니티 모듈 및 유저 모듈 리팩토링

-  커뮤니티 모듈의 컨트롤러 및 서비스 구조 개선
-  유저 모듈의 코드 정리 및 최적화
-  Prisma 클라이언트 관련 파일 업데이트
-  불필요한 코드 제거 및 주석 추가

2anizirong committed:
🔀 Merge: update with develop

2anizirong committed:
♻️ Refactor: 게시글 목록 조회에서 필터 추가

2anizirong committed:
Merge branch 'develop' into refactor/#89-community

jongse7 commented on src/modules/community/controllers/v1/community.v1.controller.ts:
```diff
@@ -73,23 +84,43 @@ export class CommunityV1Controller {
+      filter === ArticleFilterType.LIKE || // 내가 찜한 글
+      filter === ArticleFilterType.COMMENT; // 댓글 적은 글
+
+    if (requiresAuth && !userId) {
```
이거 search query도 추가돼야 할 거 같아요!! @2anizirong

2anizirong committed:
Merge branch 'develop' into refactor/#89-community

2anizirong committed:
Merge branch 'develop' of https://github.com/UMC-7th-Peekle/peekle-api-nest into refactor/#89-community

2anizirong committed:
♻️ Refactor: 커뮤니티 게시글 조회 관련 파라미터 추가
- 제목이나 내용으로 게시글을 검색할 수 있는 'search' 파라미터 추가
- 해당 게시글에 유저 작성 여부 추가
- 해당 게시글에 유저 좋아요 여부 추가

2anizirong committed:
Merge branch 'refactor/#89-community' of https://github.com/UMC-7th-Peekle/peekle-api-nest into refactor/#89-community

2anizirong committed:
♻ Refactor: 커뮤니티 게시글 찜/오너 관련 dto 추가

2anizirong committed:
Merge branch 'develop' into refactor/#89-community

jack0928 commented on src/modules/community/controllers/v1/community.v1.controller.ts:
[P3] 게시글 수정 부분에서 @ApiMultiFileAndJson과 @ApiOkResponse에서 UpdateArticleDto가 아닌 CreateArticleDto를 쓰는 이유가 뭔가요?

2anizirong replied:
두 dto 내용이 유사하다고 생각해서 그냥 하나로 통일할까 생각했는데 updateDto로 변경하는게 맞을 것 같아용 수정하겠슴당

jack0928 reviewed (changes requested):
[P2] 간단한 수정 제안사항 몇 가지 올려놨습니다. 확인하고 수정해주세요.

2anizirong committed:
♻️ Refactor: 리뷰 반영 및 DTO 수정
- likeCount랑 commentCount를 ArticleDto에 추가
- UpdateArticleDto로 수정 API 응답 타입 변경
- 컨트롤러 메서드 시그니처에서 @Body() 데코레이터 추가
- 불필요한 import 문 제거

2anizirong committed:
Merge branch 'develop' into refactor/#89-community

2anizirong committed:
Merge branch 'refactor/#89-community' of https://github.com/UMC-7th-Peekle/peekle-api-nest into refactor/#89-community

jack0928 reviewed (approved):
[P5] 수고하셨습니다.
